apps/src/code-studio/components/AudioRecorder.jsx

```diff
@@ -26,7 +26,8 @@ export default class AudioRecorder extends React.Component {
     this.slices = [];
     this.state = {
       audioName: "",
-      recording: false
+      recording: false,
+      cancelling: false
```

Comment:
Spelling: Apparently [two `l`s is the British English form](http://grammarist.com/spelling/cancel/) and it's more common in American English to write "canceling."
Naming (just discussion, not sure I'm right): I think I would have gone with "canceled" here. I don't normally think of cancel as a long-running operation, but it definitely is, as implemented here.  The other angle on it is whether we're thinking of 'cancel' as a user action or a system action.  If it's a user action I'd argue it is "canceled" as soon as the user clicks the button, but as a system action we still have to carry it out so... I don't know which is right :laughing:.  We may be able to eliminate this ambiguity altogether, see other comments.

Funny... Chrome's spell-check wants me to write "cancellation."
![image](https://user-images.githubusercontent.com/1615761/44736134-46dbc800-aaa3-11e8-9ce6-082ceacaae6b.png)